1.23.0

matteobruni opened the pull request:
feat: added Vector class taken from v2, this will be used wisely

matteobruni committed:
feat: added Vector class taken from v2, this will be used wisely

matteobruni committed:
Merge branch 'main' into v1

ghost commented:
## DeepCode's analysis on [#ebe64f](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/00b39a851c049f1ae150729ec5211c152a70fb0e/matteobruni/tsparticles/ebe64f75ff892e94b1ac605b9890ebedad95b3e8/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) found:

- :information_source: **1** minor issue. :point_down:
- :heavy_check_mark: **1** issue was fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/ebe64f75ff892e94b1ac605b9890ebedad95b3e8/core/main/src/Core/Particle/Vector.ts#L4">Vector.ts:4</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/ebe64f75ff892e94b1ac605b9890ebedad95b3e8/core/main/src/Core/Particle/Vector.ts#L8">Vector.ts:8</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/ebe64f75ff892e94b1ac605b9890ebedad95b3e8/core/main/src/Core/Particle/Vector.ts#L12">Vector.ts:12</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/00b39a851c049f1ae150729ec5211c152a70fb0e/matteobruni/tsparticles/ebe64f75ff892e94b1ac605b9890ebedad95b3e8/pr/_/%2Fcore%2Fmain%2Fsrc%2FCore%2FParticle%2FVector.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/00b39a851c049f1ae150729ec5211c152a70fb0e/matteobruni/tsparticles/ebe64f75ff892e94b1ac605b9890ebedad95b3e8/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
feat: first round of vector on particles done

matteobruni committed:
build: website updated

matteobruni committed:
build: improved confetti shape

matteobruni committed:
build: particle position is now a vector

matteobruni committed:
build: fixed tests

matteobruni committed:
build: fixed some imports

matteobruni committed:
build: some code refactoring

matteobruni committed:
build: fixed tests

matteobruni committed:
build: improved absorbers

matteobruni committed:
build: used Vector in repulser

matteobruni committed:
build: prettiefied code

matteobruni committed:
build: improved particle velocity calc

matteobruni committed:
build: dependabot security updates

matteobruni committed:
build: improved confetti shape

matteobruni committed:
build: fixed import

matteobruni committed:
build: prettiefied code

matteobruni committed:
build: fixed some imports

matteobruni committed:
build: options now can accept any value, use at own risk

matteobruni committed:
chore(release): published new version

 - ng-particles@2.6.0
 - inferno-particles@1.6.0
 - jquery-particles@1.23.0
 - preact-particles@1.23.0
 - react-tsparticles@1.23.0
 - svelte-particles@1.6.0
 - particles.vue@2.6.0
 - particles.vue3@1.6.0
 - tsparticles-editor@1.23.0
 - tsparticles@1.23.0
 - ng-particles-demo@1.8.0
 - editor-demo@1.8.0
 - inferno-particles-demo@1.6.0
 - jquery-particles-demo@1.8.0
 - tsparticles-demo@1.8.0
 - preact-particles-demo@1.8.0
 - react-tsparticles-demo@1.8.0
 - svelte-demo@1.6.0
 - particles.vue-demo@1.8.0
 - particles.vue3-demo@1.6.0
 - tsparticles-preset-60fps@1.8.0
 - tsparticles-preset-background-mask@1.8.0
 - tsparticles-preset-basic@1.8.0
 - tsparticles-preset-big-circles@1.8.0
 - tsparticles-preset-bouncing@1.8.0
 - tsparticles-preset-bubbles@1.8.0
 - tsparticles-preset-fire@1.8.0
 - tsparticles-preset-font-awesome@1.8.0
 - tsparticles-preset-fountain@1.8.0
 - tsparticles-preset-sea-anemone@1.8.0
 - tsparticles-preset-snow@1.8.0
 - tsparticles-preset-stars@1.8.0
 - tsparticles-shape-bubble@1.8.0
 - tsparticles-shape-confetti@1.8.0
 - tsparticles-shape-heart@1.8.0
 - tsparticles-shape-multiline-text@1.8.0
 - tsparticles-shape-rounded-rect@1.8.0
 - tsparticles-shape-spiral@1.8.0
 - cra-template-particles-typescript@1.6.0
 - cra-template-particles@1.6.0
 - tsparticles-website@1.23.0
